core/controllers/question_test.py

```diff
@@ -362,63 +199,79 @@ def test_get(self):
-                'cmd': collection_domain.CMD_ADD_COLLECTION_SKILL,
-                'name': 'test'
-            }], 'Add a new skill')
-        collection = collection_services.get_collection_by_id(
-            self.collection_id)
-        self.skill_id = collection.get_skill_id_from_skill_name('test')
-        collection_node = collection.get_node(self.exp_id)
-        collection_node.update_acquired_skill_ids([self.skill_id])
+        payload = {}
+        change_list = [{'cmd': 'update_question_property',
+                        'property_name': 'title',
+                        'new_value': 'ABC',
+                        'old_value': 'A Question'}]
+        payload['change_list'] = json.dumps(change_list)
+        payload['commit_message'] = 'update title'
+        self.login(self.NEW_USER_EMAIL)
+        response_json = self.put_json(
+            '%s/%s/%s' % (
+                feconf.QUESTION_DATA_URL, self.collection_id, question_id),
+            payload, expect_errors=False)
+        self.assertIn('question_id', response_json.keys())
 
-        # Update the acquired skill IDs for the exploration.
-        collection_services.update_collection(
-            self.new_user_id, self.collection_id, [{
-                'cmd': collection_domain.CMD_EDIT_COLLECTION_NODE_PROPERTY,
-                'property_name': (
-                    collection_domain.COLLECTION_NODE_PROPERTY_ACQUIRED_SKILL_IDS), # pylint: disable=line-too-long
-                'exploration_id': self.exp_id,
-                'new_value': [self.skill_id]
-            }], 'Update skill')
+        del payload['change_list']
+        self.put_json(
+            '%s/%s/%s' % (
+                feconf.QUESTION_DATA_URL, self.collection_id,
+                question_id), payload, expect_errors=True,
+            expected_status_int=404)
 
-        self.payload = {}
+        del payload['commit_message']
+        payload['change_list'] = json.dumps(change_list)
+        self.put_json(
+            '%s/%s/%s' % (
+                feconf.QUESTION_DATA_URL, self.collection_id,
+                question_id), payload, expect_errors=True,
+            expected_status_int=404)
 
+        payload['commit_message'] = 'update title'
+        self.put_json(
+            '%s/%s' % (feconf.QUESTION_DATA_URL, self.collection_id),
+            payload, expect_errors=True,
+            expected_status_int=404)
 
-    def test_black_box(self):
-        question = question_domain.Question(
-            'dummy', 'A Question',
-            exp_domain.State.create_default_state('ABC').to_dict(), 1,
-            self.collection_id, 'en')
-        self.payload['question'] = question.to_dict()
-        self.payload['skill_id'] = self.skill_id
+        self.logout()
+        response = self.testapp.put(
+            '%s/%s/%s' % (
+                feconf.QUESTION_DATA_URL, self.collection_id, question_id),
+            payload, expect_errors=True)
+        self.assertEqual(response.status_int, 302)
```

Comment:
This is weird -- 302 is not an error. Shouldn't it just be expected_status_int=302?

Reply:
updated in 8ec2d3e.